client/client.go

```diff
@@ -2847,12 +2847,15 @@ func (c *Client) setGaugeForUptime(hStats *stats.HostStats, baseLabels []metrics
 func (c *Client) emitHostStats() {
 	nodeID := c.NodeID()
 	hStats := c.hostStatsCollector.Stats()
-	node := c.Node()
 
-	node.Canonicalize()
+	c.configLock.RLock()
+	nodeStatus := c.configCopy.Node.Status
+	nodeEligibility := c.configCopy.Node.SchedulingEligibility
+	c.configLock.RUnlock()
```

Comment:
c.configCopy.Node.Status is written to by the node registration/heartbeating goroutine. The `c.Node()` helper only serializes access to the `c.configCopy` pointer, not the fields within it, so we need to hold the lock while accessing the fields.

I think the c.config vs c.configCopy situation could be improved, but I just wanted to get my tests `-race` clean.